astronomer/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -179,24 +181,40 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
         )
         if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
             raise AirflowException(f"Sagemaker transform Job creation failed: {response}")
-        else:
-            end_time: Optional[float] = None
-            if self.max_ingestion_time is not None:
-                end_time = time.time() + self.max_ingestion_time
-            self.defer(
-                timeout=self.execution_timeout,
-                trigger=SagemakerTrigger(
-                    poke_interval=self.check_interval,
-                    end_time=end_time,
-                    aws_conn_id=self.aws_conn_id,
-                    job_name=transform_config["TransformJobName"],
-                    job_type="Transform",
-                    response_key="TransformJobStatus",
-                ),
-                method_name="execute_complete",
-            )
 
-    def execute_complete(self, context: "Context", event: Dict[str, Any]) -> Dict[str, Any]:
+        response = self.hook.describe_transform_job(transform_config["TransformJobName"])
+        status = response["TrainingJobStatus"]
+        if status in self.hook.failed_states:
+            raise AirflowException(f"SageMaker job failed because {response['FailureReason']}")
+
+        if status == "Completed":
+            self.log.info(f"{self.task_id} completed successfully.")
+            transform_config = self.config.get("Transform", self.config)
```

Comment:
Do we need this line to get `transform_config`, we already have this somewhere above, no?

Reply:
Yep, miss that one. Just removed it. Thanks!